https://github.com/grpc-ecosystem/grpc-opentracing/issues/15

hzariv opened the pull request:
https://github.com/grpc-ecosystem/grpc-opentracing/issues/15
https://github.com/grpc-ecosystem/grpc-opentracing/issues/17
https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

hzariv committed:
https://github.com/grpc-ecosystem/grpc-opentracing/issues/15
https://github.com/grpc-ecosystem/grpc-opentracing/issues/17
https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

hzariv committed:
https://github.com/grpc-ecosystem/grpc-opentracing/issues/15
https://github.com/grpc-ecosystem/grpc-opentracing/issues/17
https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

hzariv committed:
https://github.com/grpc-ecosystem/grpc-opentracing/issues/15
https://github.com/grpc-ecosystem/grpc-opentracing/issues/17
https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

bhs commented on java/src/main/java/io/opentracing/contrib/ServerTracingInterceptor.java:
```diff
@@ -83,8 +82,15 @@ public ServerServiceDefinition intercept(BindableService bindableService) {
         for (String key : headers.keys()) {
-            String value = headers.get(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER));
-            headerMap.put(key, value);
+            if (key.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
```
I'd argue that we should just drop these... WDYT? Especially since `byteValue` may not even be valid `UTF-8`, thus leading to undefined behavior.

I greatly dislike how the OT user now needs to "know" about this weird `-bin` suffix semantic, btw... but not sure how to document that. It's some comfort that, in practice, it's unlikely to crop up. But only because of its obscurity.

hzariv replied:
If you think binary headers provide no value, then we can drop them. gRPC uses binary header to serialize some context. If it doesn't add value for tracing, I am all for dropping them.

bhs commented on java/src/main/java/io/opentracing/contrib/ClientTracingInterceptor.java:
```diff
@@ -299,7 +299,7 @@ public ClientTracingInterceptor build() {
     }
 
-    enum ClientRequestAttribute {
+    public enum ClientRequestAttribute {
```
You don't need to make this change, but FYI I regret this approach... would be less code and more flexibility to do a `SpanDecorator` here like we did in Go. https://github.com/grpc-ecosystem/grpc-opentracing/blob/master/go/otgrpc/options.go#L41

In case you feel inspired, we could add something similar as an alternative.

hzariv replied:
This change is for https://github.com/grpc-ecosystem/grpc-opentracing/issues/17 and https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

bhs commented on java/build.gradle:
```diff
@@ -47,16 +47,16 @@ repositories {
-	testCompile 'io.grpc:grpc-stub:1.0.0-pre1'
+	compile 'io.grpc:grpc-core:1.1.2'
+	compile 'io.opentracing:opentracing-api:0.15.0'
+    testCompile 'io.opentracing:opentracing-mock:0.15.0'
```
nit: some weird indenting thing happened here.

hzariv replied:
fixed

bhs commented on java/README.rst:
```diff
@@ -16,15 +16,15 @@ This package is available on Maven Central and can be added to your project as f
             <groupId>io.opentracing.contrib</groupId>
             <artifactId>grpc-opentracing</artifactId>
-            <version>0.1.0</version>
+            <version>0.2.0</version>
```
the minor version bump is for the `public`s, yes? (SGTM)

hzariv replied:
@bhs yes API was broken in 0.1.0

hzariv committed:
https://github.com/grpc-ecosystem/grpc-opentracing/issues/15
https://github.com/grpc-ecosystem/grpc-opentracing/issues/17
https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

hzariv committed:
https://github.com/grpc-ecosystem/grpc-opentracing/issues/15
https://github.com/grpc-ecosystem/grpc-opentracing/issues/17
https://github.com/grpc-ecosystem/grpc-opentracing/issues/18

bhs reviewed (approved):
Looks great. Thanks again.

bhs commented:
@hzariv there's a greater than 50% chance that I managed to get `gradle uploadArchives` to work... basically a bunch of yak-shaving with GPG, but in the end I got this in the shell:

```
$ gradle uploadArchives
:compileJava
Download https://repo1.maven.org/maven2/io/grpc/grpc-core/1.1.2/grpc-core-1.1.2.pom
Download https://repo1.maven.org/maven2/com/google/guava/guava/20.0/guava-20.0.pom
Download https://repo1.maven.org/maven2/com/google/guava/guava-parent/20.0/guava-parent-20.0.pom
Download https://repo1.maven.org/maven2/io/grpc/grpc-context/1.1.2/grpc-context-1.1.2.pom
Download https://repo1.maven.org/maven2/com/google/instrumentation/instrumentation-api/0.3.0/instrumentation-api-0.3.0.pom
Download https://repo1.maven.org/maven2/io/grpc/grpc-core/1.1.2/grpc-core-1.1.2.jar
Download https://repo1.maven.org/maven2/com/google/guava/guava/20.0/guava-20.0.jar
Download https://repo1.maven.org/maven2/io/grpc/grpc-context/1.1.2/grpc-context-1.1.2.jar
Download https://repo1.maven.org/maven2/com/google/instrumentation/instrumentation-api/0.3.0/instrumentation-api-0.3.0.jar
:processResources UP-TO-DATE
:classes
:jar
:javadoc
:javadocJar
:sourcesJar
:signArchives
:uploadArchives

BUILD SUCCESSFUL

Total time: 20.434 secs
```

hzariv commented:
@bhs thanks very much for review and merge. Do you have and ETA when 0.2.0 artifact will be released to maven central? (we use maven central repo for this artifact)

bhs commented:
@hzariv it should have happened already. something must be broken with the gradle config... argh. Not my area of expertise to say the least... I'll take a look, though.

bhs commented:
Some of the changes in https://github.com/grpc-ecosystem/grpc-opentracing/pull/21 may have fixed this. There's some lag in the central repo, but in an hour we'll have a better sense.
